Introduced the concept of organizations to the system.

lmartin73 opened the pull request:
Defined the organization class and updated the user profile to
included a list of organizaitons that are awaiting confirmation and
a list of organizations to which the user is a member.

lmartin73 committed:
Introduced the concept of organizations to the system.

Defined the organization class and updated the user profile to
included a list of organizaitons that are awaiting confirmation and
a list of organizations to which the user is a member.

lmartin73 committed:
Removed organization debug comments and routines

jhubbard2017 commented on src/app/_be/auth/useracct.js:
```diff
@@ -237,6 +239,31 @@ export class UserProfile {
 
+    getOrganizations(){
+        var memberOrgs = [];
+        for (ordUID in this.organizations){
```
ordUID or orgUID?

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/auth/useracct.js:
```diff
@@ -237,6 +239,31 @@ export class UserProfile {
+    getPendingOrganizations(){
+
+        var pendingOrgs = [];
+        for (ordUID in this.organizations){
```
ordUID or orgUID?

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -166,30 +144,32 @@ export default class Organization {
     addMember(userUID=null){
         if(userUID && this.uid){
```
Check this as well for add administrator (like 94)

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -46,94 +46,72 @@ export default class Organization {
 
     //make user an administrator of this organization
     addAdministrator(userUID=null){
         if(userUID){
```
Check this.uid as well

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -203,27 +183,28 @@ export default class Organization {
     denyMemberRequest(userUID=null){
         if(userUID){
```
check this.uid here as well

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -203,27 +183,28 @@ export default class Organization {
 
     setOwner(userUID=null){
-       if(userUID){
+       if(userUID != null){
```
check this.uid here

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -203,27 +183,28 @@ export default class Organization {
     setOwner(userUID=null){
-       if(userUID){
+       if(userUID != null){
+            console.log(userUID)
```
remove log!

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -247,19 +230,100 @@ export default class Organization {
     requestMemembership(userUID=null){
         if(userUID){
```
check this.uid here

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -247,19 +230,100 @@ export default class Organization {
+        /* set organization data for the owner */
+        this.setOwner(orgData.ownerUID)
+
+        /* update user's profile to reflect ownership of organization
```
Remove code, since you've already set the owner above.

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -247,19 +230,100 @@ export default class Organization {
+                fax: snapshot.val().fax ? snapshot.val().fax : "" ,
+                logoURL: snapshot.val().logoURL ? snapshot.val().logoURL : ""
+            }
+            this.administrators = snapshop.val().administrators ? snapshop.val().administrators : "" ;
```
If null, this should be an empty object instead of empty string

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -247,19 +230,100 @@ export default class Organization {
+                logoURL: snapshot.val().logoURL ? snapshot.val().logoURL : ""
+            }
+            this.administrators = snapshop.val().administrators ? snapshop.val().administrators : "" ;
+            this.members = snapshot.val().members ? snapshot.val().members : "";
```
If null, this should be an empty object instead of empty string

lmartin73 replied:
done

jhubbard2017 commented on src/app/app.js:
```diff
@@ -5,8 +5,8 @@ import {syncHistoryWithStore, routerActions} from 'react-router-redux'
-
-import {logoutUser, startListeningToAuth} from './components/user'
+import {UserProfile} from '_be/auth/useracct'
+import {logoutUser, startListeningToAuth, attemptLoginWithEmail} from './components/user'
```
Don't need attemptLoginWithEmail

lmartin73 replied:
done

jhubbard2017 commented on src/app/app.js:
```diff
@@ -5,8 +5,8 @@ import {syncHistoryWithStore, routerActions} from 'react-router-redux'
 import {FIRconfig} from './config/firebaseConfig' // import firebase config details
-
-import {logoutUser, startListeningToAuth} from './components/user'
+import {UserProfile} from '_be/auth/useracct'
```
Don't need UserProfile

lmartin73 replied:
done

jhubbard2017 commented on src/app/_be/organizations/Organization.js:
```diff
@@ -46,94 +46,72 @@ export default class Organization {
 
     constructor(orgID = null){
```
This class needs a method to set the organization uid (If not provided initially), and populates the organization class with the correct organization data from the database

lmartin73 committed:
Pull Request #30

Made changes to correct code bugs in previous commit. Added the setOrg
method to the Organizaiton class.

lmartin73 committed:
More updates to re mitage errors when downloading Organizaiton info

jhubbard2017 commented:
👍

jhubbard2017 commented:
💯
